src/util/replace_symbol.cpp

```diff
@@ -89,33 +91,75 @@ bool replace_symbolt::replace(exprt &dest) const
   if(!have_to_replace(dest))
     return result;
 
-  if(dest.id()==ID_symbol)
+  if(dest.id()==ID_member)
+  {
+    member_exprt &me=to_member_expr(dest);
+
+    if(!replace(me.struct_op(), replace_with_const))
+      result=false;
+  }
+  else if(dest.id()==ID_index)
+  {
+    index_exprt &ie=to_index_expr(dest);
+
+    if(!replace(ie.array(), replace_with_const))
+      result=false;
+
+    if(!replace(ie.index()))
```

Comment:
@martin-cs: Exactly.

@tautschnig: The index is always replaced here. The ``if`` is there to check the return value of ``replace()`` which indicates whether the expression was simplified.